### DISCUSSION TITLE
Add environment to Align Columns

### DISCUSSION THREAD

**Discussion by fernandozucatelli** [texstudio-org/texstudio#1579] [Ideas]
I would like to know if it is possible to add an environment to the environments that may go under the "Align Columns" procedure.

This button is magical, i really enjoy how it aligns **tabular**, **array**, **matrix** (and variants), but, when I try to apply it to some new environment, regardless if I created it myself or from another package, that is some variation from previously cited environments, the operation just don't work.

A simple workaround I've been using is to replace the new environment's name for one that "Align Columns" work, applying it, and then replacing the environment's name back.

By the way, it would also be nice a global alignment option, something like I did [here](https://tex.stackexchange.com/questions/572259/how-to-create-a-texstudio-macro-to-apply-align-columns-for-all-matrices-i-want) by creating my persona not-so-neat macro.

---------------------- Edit

Let's say I created my own array:
`\newenvironment{myArray}{\begin{pmatrix}}{\end{pmatrix}}`

And the I inserted some non-aligned numbers
```
\begin{equation}
  \begin{myArray}
    1 & 2    &  3        & 7 \\
    4 & 5        &   6 & 8 
  \end{myArray}
\end{equation}
```

This environment can't have the &s aligned through the command "Align Columns".
To have it so, I must replace its name from "myArray" to "array", **then** apply "Align Columns" and **then** rename it back to "myArray".

It would be nice to **add** my own environment "myArray" to the set of environments the command "Align Columns" can be applied to.

**Comment by benibela** [comment]
Another workaround would be to put a  cursor mirror at each & with alt+ctrl+click and use Edit / Line Operations / Align Cursors

**Reply by fernandozucatelli** [reply]
Well, this trick do align ampersands but then I should click in all of them, which is not a practical operation.
As I can create a keyboard shortcut for "Align Columns" button, I can easy place the mouse inside any environment that may have "&" and/or "\\", press the shortcut combination, and like magic, the command aligns all elements.
What I really would like to do is to expand the set of environments the command "Align Columns" works.

**Comment by sunderme** [comment]
please open an issue/feature request
